packages/next/src/build/index.ts

```diff
@@ -645,6 +645,7 @@ async function writeStandaloneDirectory(
         const middlewareOutput = path.join(
           distDir,
           STANDALONE_DIRECTORY,
+          path.relative(outputFileTracingRoot, requiredServerFiles.appDir),
```

Comment:
Probably better to use `appDir` and remove `requiredServerFiles.config.distDir,` below